app/Http/Controllers/SettingsController.php

```diff
@@ -405,16 +405,22 @@ public function add_support(Request $request)
                         }
                     }
             }
-            /** check point if removing support from one camp and adding in another 
+            /** check point 1 if removing support from one camp and adding in another 
              * in that case delegate should not promted it should simply shifted with DS
             */
-            if(count($data['support_order']) > 0){
+            $promoteDelegate = true;
+            if(isset($data['support_order']) && count($data['support_order']) > 0){
                 $promoteDelegate = false;
-            }                    
+            }
+            /** check point 2 if all suppot removed  */
+            $ifSupportLeft = true;
+            if(isset($data['removed_camp']) && count($mysupports) == count($data['removed_camp'])){
+                $ifSupportLeft = false;
+            }
```

Comment:
Probably will be replaced with 

`$ifSupportLeft = ( isset($data['removed_camp']) && count($mysupports) == count($data['removed_camp']) ) ? true : false;`